Whats the Android way?

Hi there,

in your documentation I can read:

    protobuf {
        plugins {
            javalite {
                artifact = 'com.google.protobuf:protoc-gen-javalite:3.0.0'
            }
        }
        generateProtoTasks {
            all().each { task ->
                task.plugins {
                    javalite { }
                }
            }
        }
    }

But using Android Studio 3.0 Beta 7 with a default project and an empty activity I geht this error:

	FAILURE: Build failed with an exception.

	* What went wrong:
	Execution failed for task ':app:generateDebugProto'.
	> java.io.IOException: Cannot run program "protoc": CreateProcess error=2, Das System kann die angegebene Datei nicht finden

	* Try:
	Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

	* Get more help at https://help.gradle.org

	BUILD FAILED in 3s

So using Google I found [this](https://github.com/google/protobuf-gradle-plugin/issues/89) solution somewhere in your issues:

	protobuf {
        protoc {
            artifact = "com.google.protobuf:protoc:3.0.0"
        }
        plugins {
            lite {
                artifact = "com.google.protobuf:protoc-gen-javalite:3.0.0"
            }
        }
        generateProtoTasks {
            all().each { task ->
                task.builtins {
                    remove java
                }
                task.plugins {
                    lite { }
                }
            }
        }
    }

So what is the correct way to use protobuf with Android ?

BTW: I use "com.android.tools.build:gradle:3.0.0-rc2" and Gradle 4.1

Cheers,
DaRolla

--- a/README.md
+++ b/README.md
@@ -340,6 +340,10 @@ dependencies {
 }
 
 protobuf {
+  protoc {
+    // You still need protoc like in the non-Android case
+    artifact = 'com.google.protobuf:protoc:3.0.0'
+  }
   plugins {
     javalite {
       // The codegen for lite comes as a separate artifact
@@ -348,6 +352,11 @@ protobuf {
   }
   generateProtoTasks {
     all().each { task ->
+      task.builtins {
+        // In most cases you don't need the full Java output
+        // if you use the lite output.
+        remove java
+      }
       task.plugins {
         javalite { }
       }

